pandas/tests/extension/test_integer.py

```diff
@@ -249,6 +249,51 @@ class TestBooleanReduce(base.BaseBooleanReduceTests):
     pass
 
 
+class TestNumericAccumulation(base.BaseNumericAccumulateTests):
+    def check_accumulate(self, s, op_name, skipna):
+        # overwrite to ensure pd.NA is tested instead of np.nan
+        # https://github.com/pandas-dev/pandas/issues/30958
+        if op_name == "cumsum":
+            if s.dtype.name.startswith("U"):
+                expected_dtype = "uint64"
+            else:
+                expected_dtype = "int64"
+            result = getattr(s, op_name)(skipna=skipna)
+            expected = pd.Series(
+                integer_array(
+                    getattr(s.astype("float64"), op_name)(skipna=skipna),
+                    dtype=expected_dtype,
+                )
+            )
+            tm.assert_series_equal(result, expected)
+        elif op_name in ["cummax", "cummin"]:
```

Comment:
please try to make this simpler (the entire function), you are repeating lots of things